### ISSUE TITLE
[FRAME] Remove `is_inherent_required`

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#3662]
Currently there are two functions in the `ProvideInherent` trait that can return conflicting information to FRAME Executive; `create_inherent` and `is_inherent_required`.  
For example `is_inherent_required` can return true, while `create_inherent` returns `None`. In the same manner, `is_inherent_required` can return false but `create_inherent` `Some`.  
It also does not specify *which* inherent is required - just any.

This is creating an additional layer of indirection instead of letting `create_inherent` determine the inclusion criteria of an inherent. I think the outcome of the last discussion with @bkchr was to deprecate/remove `is_inherent_required`.  This would mean that `on_finalize` of each pallet must check the inclusion criteria manually.

**Comment by chungquantin** [comment]
Would love to discuss more for further context. So let say we refractor the `is_inherent_required`, all the pallets in `polkadot-sdk` must have the updated `on_finalize` implementation or are there any better way to handle this?

**Comment by bkchr** [comment]
> I think the outcome of the last discussion with @bkchr was to deprecate/remove `is_inherent_required`. This would mean that `on_finalize` of each pallet must check the inclusion criteria manually.

No, we should require that each inherent is always there. Stuff that isn't an inherent, shouldn't be put into an inherent. However, people still have the possibility to wrap the inners of the inherent in an `Option` and pass `None` if there is no data to construct the inherent.

**Comment by ggwpez** [comment]
> No, we should require that each inherent is always there. Stuff that isn't an inherent, shouldn't be put into an inherent. However, people still have the possibility to wrap the inners of the inherent in an `Option` and pass `None` if there is no data to construct the inherent.

You mean that `create_inherent` should rather return `RuntimeCall`? (we could first deprecate that old version and add a new one to not break code immediately)  

The API still is still not really clean since `create_inherent` is not bound to return a specific call... it just requires that it returns *a* call.
